app/models/datadog.rb

```diff
@@ -6,7 +6,10 @@ class Datadog < ApplicationRecord
   validates :api_key, presence: true
   validates :application_key, presence: true
   validates :subdomain, presence: true
-  
+
   validates_with DatadogValidator
 
+  after_save_commit do
+    DatadogSyncJob.perform_now({ api_key: api_key, application_key: application_key, subdomain: subdomain, connection_id: self.cred.connection_id })
```

Comment:
Should it be perform_async? If you use perform_now, does it run the job in the main thread? Let's re-confirm this

Reply:
Also you do not need to pass too much data into our job. We just need an ID, then within the job, we can query for api_key, application_key, subdomain, etc... from the data base